### ISSUE TITLE
How to use Custom Index

### ISSUE THREAD

**Issue by jobs-git** [10up/ElasticPress#1032]
The Title says it all.

**Comment by nprasath002** [comment]
If you want a different index name you can use `ep_index_name` filter

**Comment by jobs-git** [comment]
I did, I used the index created by logstash but when I test it, nothing shows up in the results.

**Comment by nprasath002** [comment]
Unfortunately we don't support logstash related issues. Let me know if you have any questions with ElasticPress API

**Comment by brandwaffle** [comment]
@jobs-git ElasticPress will try to delete and re-create whatever index you specify as a name. Are you using the logstash index because the Elasticsearch server won't allow programmatic creation/deletion of an index? If so, ElasticPress won't work out of the box because it does need to delete/create the index.

**Comment by jobs-git** [comment]
Hi, Elasticpress has difficulty in indexing the database containing more than 200k rows, it stops at less than 100k. Moreover the speed at which elasticpress indexes the database seems to slow down as the number of entries (it has indexed) increases, it took maybe 27mins before stopping abruptly . On the other hand, logstash can just do it in seconds.

**Comment by irakligeek** [comment]
I would really appreciate if someone will explain how to use custom index for ElasticPress or any if anyone point me to a documentation related to this issue. Thanks

**Comment by chersull** [comment]
Hi there, I realize this post is pretty old, but just putting this out there.  I was able to apply a custom index by doing the following. Not sure if all these steps are necessary, but it took me a while to figure it out so hopefully this will save someone else some time.  

1. Find the `put_mapping` function in the plugins\elasticpress\classes\class-ep-api.php file.  Figure out which mappings file your installation is using based on your version of Elasticsearch as documented in this function.
2. Place a copy of that file (found in plugins\elasticpress\includes\mappings, ours is pre-5-0.php) somewhere inside of your Wordpress file structure, renaming it to something unique.  Make your custom changes to that file. Keeping your custom file outside of the plugins directory will keep it from being deleted when you upgrade the plugin.   
3. To have this mapping pulled in from your ElasticPress plugin dashboard when reindexing, use the ep_config_mapping_file filter - 

`add_filter( 'ep_config_mapping_file', array( $this, 'adjust_ep_mapping_args' ), 10, 1 );`

4. The function this calls is simply - 
```
public function adjust_ep_mapping_args( $args ) {
      $args = /path_to_directory/pre-5-0_amended.php';
      return $args;
}
```
5. I had to reindex by first deleting the existing index via a curl command, as seem below.  Then I used the reindex button on the ElasticPress dashboard, and my index was rebuilt using my custom index file.
`curl -X DELETE '/index_name'`
